### DISCUSSION TITLE
VS Extension: Run/Debug test from feature file

### DISCUSSION THREAD

**Discussion by TobiasWenzel** [reqnroll/Reqnroll#270] [Ideas]
Visual Studio has a small feature that is extremely useful: to run or debug a test which you have currently opened in the editor.

I would like to have this feature also for the editor of the .feature file. So that when my cursor is located in a line of a scenario, I can **type the keyboard shortcut to immediately debug that test**.
Also an entry to the context menu should be added to do that.

It would be perfect if that feature could use the same keyboard shortcuts as the appropriate VS commands for running (`Ctrl-R`, `T`) and debugging (`Ctrl-R`, `Ctrl-T`) a test.

**Comment by gasparnagy** [comment]
Yes, this would be really great.
A few years ago I spent a couple of days figuring out how to get this working, but at that time it finally turned out that Visual Studio enables these features only for `.cs` (and some other) files and this extension was hard-coded in the Visual Studio code.

I was also thinking on hooking on the command, quickly switch to the C# code behind file and replay the command there, but this was very brittle. So I gave up.

I don't know if there has been any change in that regards, but I'm not very optimistic. If someone has any other idea how this could be supported then let's give it a try.

I use NCrunch, that is a commercial test runner tool and that is able to hook this into the feature file without any extra code on our side.

![image](https://github.com/user-attachments/assets/677609c5-67d6-4b15-8351-5765ffd07e8f)

Based on that maybe we should try submitting this to the Visual Studio team, maybe they could do something for it, if we add enough votes to the issue.
